### PR TITLE
Redundant .c_str() usage in rclcpp_components triggers ament_clang_tidy warning

### PR DESCRIPTION
**Pull request by LihanChen2004**
The code snippet:

```cpp
std::string name = clazz.c_str();
```
is redundant and unnecessary when initializing a `std::string` object. 

The method `std::string::operator=(const char*)`, which takes a C-style string (`const char*`), introduces an extra conversion step when the input is already a `std::string`. Specifically, the `c_str()` method is used to extract a `const char*` pointer from `clazz`, and then `std::string` reconstructs the string from the pointer.

Instead, the simpler and more efficient approach is:

```cpp
std::string name = clazz;
```

This directly utilizes the copy constructor or move constructor of `std::string`, avoiding the unnecessary intermediate conversion to `const char*`. It results in cleaner, more idiomatic C++ code with no performance penalty.

Addresses: https://github.com/ros2/rclcpp/issues/2717

### CODE REVIEW

File: rclcpp_components/src/node_main.cpp.in [hunk 1]
```diff
@@ -40,7 +40,7 @@ int main(int argc, char * argv[])
   auto loader = new class_loader::ClassLoader(library_name);
   auto classes = loader->getAvailableClasses<rclcpp_components::NodeFactory>();
   for (const auto & clazz : classes) {
-    std::string name = clazz.c_str();
+    std::string name = clazz;
     if (name.compare(class_name) == 0) {
```
Comment (fujitatomoya) [inline rclcpp_components/src/node_main.cpp.in]: Do we actually need this `name` local variable under for loop statement? can we just do the following instead?

```suggestion
    if (clazz.compare(class_name) == 0) {
```

**Review by fujitatomoya — COMMENTED** [review]
@LihanChen2004 thanks for creating PR.

File: rclcpp_components/src/node_main.cpp.in (hunk 1 above)
Comment (LihanChen2004) [reply rclcpp_components/src/node_main.cpp.in]: I think it is much more better :)

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
Pulls: https://github.com/ros2/rclcpp/pull/2718
Gist: https://gist.githubusercontent.com/clalancette/db85df4ed40b52a0cb3945c91d45b99b/raw/92681d90a6dba181cd4c37559e74d034beb99aa0/ros2.repos
BUILD args:  --packages-above-and-dependencies rclcpp_components
TEST args:  --packages-above rclcpp_components
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/15015
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/22331/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/16594/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1862/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/23122/)

**Review by fujitatomoya — APPROVED** [review]
lgtm, CI failing because this targets to humble only. i will restart.

**Comment by fujitatomoya** [comment]
Pulls: ros2/rclcpp#2718
Gist: https://gist.githubusercontent.com/fujitatomoya/3d68220de5abb05de1c42d5264ce9e62/raw/fcdee4cfd58f36c0f79f451e2de5284503621470/ros2.repos
BUILD args:  --packages-above-and-dependencies rclcpp_components
TEST args:  --packages-above rclcpp_components
ROS Distro: humble
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/15016
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/22332/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/16595/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1863/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/23123/)

**Pull request merged**